pkg/controller/types.go

```diff
@@ -85,12 +88,22 @@ func NewVarnishController(
 	}, nil
 }
 
+func getEnvironment() map[string]string {
+	items := make(map[string]string)
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		items[pair[0]] = pair[1]
```

Comment:
I'd preffer checking `if len(pair) > 1` - just to ensure that there are no explosions.

Reply:
This actually doesn't fix the explosion. Environment variable is always at the format `A=B` and even with the empty value it's `A=`. The explosion happens when the template engine is accessing to the `Env` variable but can't find the key from that. So missing key -> No work. And IMHO it should be so.